Library/Homebrew/test/cli_parser_spec.rb

```diff
@@ -0,0 +1,54 @@
+require_relative "../cli_parser"
+
+describe Homebrew::CLI::Parser do
+  describe "test switch options" do
+    subject(:parser) {
+      described_class.new do
+        switch "--verbose",      "Flag for verbosity"
+        switch "--more-verbose", "Flag for higher verbosity"
+      end
+    }
+    it "with a valid option passed" do
+      args = parser.parse(["--verbose"])
+      expect(args.verbose?).to be true
+    end
+
+    it "with valid option and unnamed args passed" do
+      args = %w[--verbose unnamed args]
+      parser.parse(args)
+      expect(args).to eq %w[unnamed args]
+    end
+
+    it "with a single option passed" do
+      args = parser.parse(["--verbose"])
+      expect(args.verbose?).to be true
+      expect(args.more_verbose?).to be nil
+    end
+
+    it "with an invalid switch passed" do
+      expect { parser.parse(["--random"]) }.to raise_error(OptionParser::InvalidOption, /--random/)
+    end
+  end
+
+  describe "test long flag options" do
+    subject(:parser) {
+      described_class.new do
+        flag        "--filename", "Name of the file", required: true
+        comma_array "--files",    "Comma separated filenames"
+      end
+    }
+    it "with a long flag option and its parameter" do
+      args = parser.parse(["--filename=random.txt"])
+      expect(args.filename).to eq "random.txt"
+    end
+
+    it "with flag with its required parameter not passed" do
+      expect { parser.parse(["--filename"]) }.to raise_error(OptionParser::MissingArgument, /--filename/)
+    end
+
+    it "with comma array flag" do
```

Comment:
These test names could be improved. They should read like English sentences, including the `it`, eg `it "parses a comma array"`.